search/arch_draft.py

```diff
@@ -0,0 +1,82 @@
+"""
+This is temporary module with the arch concept for the search.
+
+@todo #331:60m  Implement the new arch at the search app.
+ All you need is to implement ContextsStack class
+ and place the other classes to the relevant places.
+ `search.search` module will be fully rewritten.
+ You also can drop away starting_by_term functionality.
+ See "is_name_start_by_term" aggregation parameter
+ inside search.search.search function.
+"""
+
+import abc
+from itertools import chain
+
+from django.db.models import QuerySet
+from django.http import JsonResponse
+
+from images.models import Image
+from pages import typing
+
+
+class Representable(abc.ABC):
+    @property
+    def name(self):
+        raise NotImplemented()
+
+    @property
+    def url(self):
+        raise NotImplemented()
+
+
+class Result:
```

Comment:
It seems this is zero-logic class, so we don't need it

Reply:
@artemiy312 , there is it's logic:
```
        self.image: Image = getattr(item, 'image', None)
        self.price: float = getattr(item, 'price', 0.0)
```

In case we don't need getattr behaviour, it should be just data type, of couse. I should use `@dataclass` deco in this case

Reply:
in general responsibility of this adapt is to cast object came from QS to the results item with clear interface. "adapt" is it's behaviour

Reply:
seems our Objects Master loves Adatper pattern =)
https://www.yegor256.com/2016/02/03/design-patterns-and-anti-patterns.html

Reply:
It is [DTO](https://en.wikipedia.org/wiki/Data_transfer_object) with zero _business_ logic.

Reply:
But isn't so important.
I don't understand how will we utilize this class and what kind of problem it will solve for us. 
We have deal with heterogeneous data, that can't be generalized in single class.

Reply:
@artemiy312 , business the logic below or not, we can't put it to a data structure.

> what kind of problem it will solve for us

it makes search results interface clear. Now it's not. Template receives a heterogeneous list where every element has 10-30 different fields